statement_parser.go

```diff
@@ -406,9 +407,42 @@ func removeStatementHint(sql string) string {
 	return sql
 }
 
+type statementsCacheEntry struct {
+	sql    string
+	params []string
+	info   *statementInfo
+}
+
+const statementCacheSize = 1000
```

Comment:
nit: should it be configurable via connection string param? If someone uses a bigger machine and can afford more entries in the cache, please ignore if you think having more entries has less benefit.

Reply:
Yes, that is a good point. It is being made configurable in one of the PRs further down this chain.